pkg/networkservice/common/begin/event_factory.go

```diff
@@ -97,6 +97,7 @@ func (f *eventFactoryClient) Request(opts ...Option) <-chan error {
 				defer cancel()
 				_, _ = f.client.Close(ctx, request.GetConnection(), f.opts...)
 				if request.GetConnection() != nil {
+					request.MechanismPreferences = nil
```

Comment:
As I can see we'll miss the [client's mechanism](https://github.com/networkservicemesh/cmd-nsc-init/blob/main/main.go#L186-L188) with this patch.

Am I missing something?

Reply:
You are right
This PR is incorrect